### PR TITLE
fix postgresql configuration and access resources

### LINKED ISSUE #472: PostgreSQL Config Resource
## Feature

As a postgreSQL admin
I want my postgreSQL server configured the same reliably
And no matter which platform I chose listed in the supported platforms it should work
So that I can have some level of configuration disaster recovery

## Supporting Information

None yet. 

## Related Issues

- #310 
- #329

### PR DESCRIPTION
**Pull request by YpNo**
### Description

Review master version which fix postgresql configuration and access resource

### Issues Resolved

#472
#329
#310 
#466
Maybe #426
#403

### Contribution Check List

- [x] Tested on Debian7, Debian8 and CentOS7
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: libraries/resource_helpers.rb [hunk 1]
```diff
@@ -22,10 +22,27 @@ module Helpers
...
-        "/var/lib/postgresql/#{new_resource.version}/main"
+        "/var/lib/postgresql/#{node.run_state['postgresql']['version']}/main"
       end
     end
+
+    def conf_dir
+      case node['platform_family']
+      when 'rhel', 'fedora', 'amazon'
+        "/var/lib/pgsql/#{node.run_state['postgresql']['version']}/data"
+      when 'debian', 'ubuntu'
```
Comment (tas50) [inline libraries/resource_helpers.rb]: no need for ubuntu here. Debian platform family includes ubuntu

File: resources/access.rb [hunk 2]
```diff
@@ -28,26 +28,25 @@
...
 action :grant do
   with_run_context :root do # ~FC037
-    edit_resource(:template, "#{postgresql_data_dir}/pg_hba.conf") do |new_resource|
+    find_resource(:service, 'postgresql')
+    edit_resource(:template, "#{conf_dir}/pg_hba.conf") do |new_resource|
       source new_resource.source
       cookbook new_resource.cookbook
       owner 'postgres'
       group 'postgres'
-      mode '0600'
```
Comment (tas50) [inline resources/access.rb]: This actually needs to be a string so we don't throw out the leading 0

File: libraries/resource_helpers.rb (hunk 1 above)
Comment (YpNo) [reply libraries/resource_helpers.rb]: Agree

File: resources/access.rb (hunk 2 above)
Comment (YpNo) [reply resources/access.rb]: I didn't know. I though it was a good practice to not use string for mode parameter

File: resources/repository.rb [hunk 3]
```diff
@@ -69,23 +69,13 @@
...
-    bash 'apt-key-add' do
-      code "sudo apt-key add #{Chef::Config[:file_cache_path]}/ACCC4CF8.asc"
-      action :nothing
-    end
-
     apt_repository 'postgresql_org_repository' do
       uri          'https://download.postgresql.org/pub/repos/apt/'
       components   ['main', new_resource.version.to_s]
       distribution "#{node['lsb']['codename']}-pgdg"
+      key new_resource.apt_gpg_key_uri
```
[the remark this answers is not in the document]
Comment (YpNo) [reply resources/repository.rb]: Wrong, I have tested with 12.14.60 version that is the requirement in README :)
But delayed_action used in cookbook depends on 12.16.+ version. I will modify the README

**Comment by YpNo** [comment]
reviewed comment Fixed ✅

**Comment by YpNo** [comment]
So, I don't understand why the build fail on Debian7 and Ubuntu1404. The service seems not running at the end...

**Comment by damacus** [comment]
I'm going to pull this in straight before mine.

**Pull request merged**